qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java

```diff
@@ -544,36 +544,66 @@ public Ask generateAskGroupUsingBaseEntity(BaseEntity baseEntity) {
 		entityMessage.setReplace(true);
 
 		// create a child ask for every valid atribute
-		baseEntity.getBaseEntityAttributes().stream()
-				.filter(ea -> defBE.containsEntityAttribute("ATT_" + ea.getAttributeCode()))
+		defBE.getBaseEntityAttributes().stream()
+//		baseEntity.getBaseEntityAttributes().stream()
+//				.filter(ea -> defBE.containsEntityAttribute("ATT_" + ea.getAttributeCode()))
 				.forEach((ea) -> {
+					String attributeCode = StringUtils.removeStart(ea.getAttributeCode(),"ATT_");
+					attributeCode = StringUtils.removeStart(attributeCode,"DEP_");
+					attributeCode = StringUtils.removeStart(attributeCode,"DFT_");
+					attributeCode = StringUtils.removeStart(attributeCode,"UNQ_");
+
+					Optional<EntityAttribute> baseEA = baseEntity.findEntityAttribute(attributeCode);
+
+					if (!baseEA.isPresent()) {
+						String questionCode = attributeCode;
+						questionCode = "QUE_" + StringUtils.removeStart(StringUtils.removeStart(questionCode, "PRI_"), "LNK_");
+						Question childQues = null;
+
+						try {
+							Attribute attribute = databaseUtils.findAttributeByCode(userToken.getRealm(), attributeCode);
```

Comment:
Use `qwandaUtils.getAttribute(attributeCode)`

Also don't catch this exception because it will hide the source of the error if the attribute does not exist

Reply:
Only create questions for ATT_xxx attributes, skip any other prefix

Reply:
Added filter for attribute only and put base entity code as target code